### DISCUSSION TITLE
Support integers exceeding Number.MAX_SAFE_INTEGER

### DISCUSSION THREAD

**Discussion by tommy-ish** [openapi-ts/openapi-typescript#1904] [Ideas]
I would like openapi-typescript and openapi-fetch to support integers exceeding Number.MAX_SAFE_INTEGER.

For example, openapi-typescript turns following schema into `bigint` type.

```yaml
type: integer
format: int64
```

openapi-fetch post following object into following JSON.

```ts
{
  foo: 9007199254740992n
}
```

```json
{
  "foo": 9007199254740992
}
```

openapi-fetch parse following JSON into following object.

```json
{
  "foo": 9007199254740992
}
```

```ts
{
  foo: 9007199254740992n
}
```

**Comment by drwpow** [comment]
This is a common sticking point with many fetch APIs. Unfortunately it balloons into having to replace the optimized, extremely-fast `response.json()` utility (that uses JS’ built-in JSON parser) with something custom, which would slow down everyone’s code (there’s no way to conditionally do it, either, because scanning for the presence of BigInt slows everything down).

Is there an existing JSON parser that support this? Any prior art of other fetch libraries that support this? Any language spec proposals that may unblock this?
